xarray/coding/cftime_offsets.py

```diff
@@ -1218,12 +1269,154 @@ def date_range(
+    | Q(E,S)-SEP | Quarter frequency, anchored at the (end, beginning) of September   |
+    +------------+--------------------------------------------------------------------+
+    | Q(E,S)-OCT | Quarter frequency, anchored at the (end, beginning) of October     |
+    +------------+--------------------------------------------------------------------+
+    | Q(E,S)-NOV | Quarter frequency, anchored at the (end, beginning) of November    |
+    +------------+--------------------------------------------------------------------+
+    | Q(E,S)-DEC | Quarter frequency, anchored at the (end, beginning) of December    |
+    +------------+--------------------------------------------------------------------+
+
+    Finally, the following calendar aliases are supported.
+
+    +--------------------------------+---------------------------------------+----------------------------+
+    | Alias                          | Date type                             | Available use_cftime=False |
+    +================================+=======================================+============================+
+    | standard, gregorian            | ``cftime.DatetimeGregorian``          | True                       |
+    +--------------------------------+---------------------------------------+----------------------------+
+    | proleptic_gregorian            | ``cftime.DatetimeProlepticGregorian`` | True                       |
+    +--------------------------------+---------------------------------------+----------------------------+
+    | noleap, 365_day                | ``cftime.DatetimeNoLeap``             | False                      |
+    +--------------------------------+---------------------------------------+----------------------------+
+    | all_leap, 366_day              | ``cftime.DatetimeAllLeap``            | False                      |
+    +--------------------------------+---------------------------------------+----------------------------+
+    | 360_day                        | ``cftime.Datetime360Day``             | False                      |
+    +--------------------------------+---------------------------------------+----------------------------+
+    | julian                         | ``cftime.DatetimeJulian``             | False                      |
+    +--------------------------------+---------------------------------------+----------------------------+
+
+    As in the standard pandas function, three of the ``start``, ``end``,
+    ``periods``, or ``freq`` arguments must be specified at a given time, with
+    the other set to ``None`` if use_cftime=False. If use_cftime=True, exactly
+    3 of the 4 can be set or 2 out of ``start``, ``end``, and``periods`` can be
+    set and ``freq`` will default to 'D'.  See the `pandas documentation
+    <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html>`_
+    for more examples of the behavior of ``date_range`` with each of the
+    parameters.
```

Comment:
The implicit default frequency is the case in pandas too, so this behavior is not unique to `use_cftime=True`.  Helpful to be clear about this though!
```suggestion
    As in the standard pandas function, exactly three of ``start``, ``end``,
    ``periods``, or ``freq`` are required to generate a date range. Note that 
    ``freq`` defaults to ``"D"`` in the event that any of ``start``, ``end``, 
    or ``periods`` are set to ``None``. See the `pandas documentation <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html>`_
    for more examples of the behavior of ``date_range`` with each of the
    parameters.
```